frame/transaction-payment/src/lib.rs

```diff
@@ -536,6 +538,10 @@ where
 		}
 	}
 
+	fn length_to_fee(length: u32) -> BalanceOf<T> {
+		T::LengthToFee::calc(&(length as u64)) // TODO: saturating / upper bound
```

Comment:
I had left this TODO here because `fn weight_to_fee` below provides an upper bound.

While I'm not quite sure why it's needed in `weight_to_fee`, it's guaranteed to be no larger than `u32::MAX` as written, which is likely much smaller than `max_block`. 

Another pair of eyes would be great though.